storage/pebble/operation/common.go

```diff
@@ -373,9 +404,17 @@ func findHighestAtOrBelow(
 			return fmt.Errorf("prefix must not be empty")
 		}
 
-		key := append(prefix, b(height)...)
+		// why height+1? because:
+		// UpperBound specifies the largest key to iterate and it's exclusive (not inclusive)
+		// in order to match all keys indexed by height that is equal to or below the given height,
+		// we could increment the height by 1,
+		// for instance, to find highest key equal to or below 10, we first use 11 as the UpperBound, so that
+		// if there are 4 keys: [prefix-7, prefix-9, prefix-10, prefix-11], then all keys except
+		// prefix-11 will be included. And iterating them in the increasing order will find prefix-10
+		// as the highest key.
+		key := append(prefix, b(height+1)...)
 		it, err := r.NewIter(&pebble.IterOptions{
-			UpperBound: append(key, ffBytes...),
+			UpperBound: key,
```

Comment:
```suggestion
			UpperBound: key,
			LowerBound: prefix,
```

Not sure, but I think it is better to constrain the keyspace for iterators where possible, since having an open iterator seems to prevent some background cleanup processes. 

[[1]](https://pkg.go.dev/github.com/cockroachdb/pebble#DB.NewSnapshot)
> Unlike the implicit snapshot maintained by an iterator, a snapshot will not prevent memtables from being released or sstables from being deleted. Instead, a snapshot prevents deletion of sequence numbers referenced by the snapshot.

[[2]](https://github.com/cockroachdb/pebble/blob/v1.1.2/iterator.go#L192)
> Iterators are expected to be short-lived (since they pin memtables and sstables)